### PR TITLE
`azurerm_search_service` - Add support for `allowed_ips`

### PR DESCRIPTION
**Pull request by mbfrahry**
```
--- PASS: TestAccAzureRMSearchService_basic (84.53s)
--- PASS: TestAccAzureRMSearchService_requiresImport (89.07s)
--- PASS: TestAccAzureRMSearchService_update (1621.21s)
--- PASS: TestAccAzureRMSearchService_ipRules (1168.95s)
```

### CODE REVIEW

File: website/docs/r/search_service.html.markdown [hunk 1]
```diff
@@ -52,6 +52,8 @@ The following arguments are supported:
 
 -> **Note:** `partition_count` and `replica_count` can only be configured when using a `standard` sku.
 
+* `ip_rules` - (Optional) A list of IP addresses that are allowed access to the search service endpoint.
```
Comment (katbyte) [inline website/docs/r/search_service.html.markdown:55]: Should we have an example of whats expected? eg `10.0.0.1` ?

File: azurerm/internal/services/search/resource_arm_search_service.go [hunk 2]
```diff
@@ -108,6 +109,15 @@ func resourceArmSearchService() *schema.Resource {
 				Default:  true,
 			},
 
+			"ip_rules": {
```
Comment (katbyte) [inline azurerm/internal/services/search/resource_arm_search_service.go:112]: would this be better as 
```suggestion
			"allowed_ips": {
```

**Review by katbyte — APPROVED** [review]
Aside from a couple comments LGMT 👍

File: website/docs/r/search_service.html.markdown (hunk 1 above)
Comment (mbfrahry) [reply website/docs/r/search_service.html.markdown:55]: We have some restrictions with loop/local addresses so I'd have to put some random IP there which doesn't feel great. I'll just update that it's a list of ipv4 addresses and we can update it further if there is confusion.

File: azurerm/internal/services/search/resource_arm_search_service.go (hunk 2 above)
Comment (mbfrahry) [reply azurerm/internal/services/search/resource_arm_search_service.go]: Done!

**Pull request merged**